### PR TITLE
Upgrade to Phlex 2

### PR DESCRIPTION
**Pull request by stephannv**
First take on upgrading the docs (I think we should rename this repo to docs, it makes more sense to me) to Phlex 2. We need https://github.com/ruby-ui/ruby_ui/pull/185 to be merged.

This PR just makes things work with Phlex 2, in the following pull requests I will reorganize some code.

### CODE REVIEW

**Review by SethHorsley — APPROVED** [review]
Great PR! If you think https://github.com/ruby-ui/ruby_ui/pull/185 is ready I approved it so you can merge it in and then we can get this merged in too.

File: Gemfile [hunk 1]
```diff
@@ -73,9 +73,10 @@ group :test do
   gem "selenium-webdriver"
 end
 
-gem "phlex-rails"
+gem "phlex", github: "phlex-ruby/phlex"
+gem "phlex-rails", github: "phlex-ruby/phlex-rails"
 
-gem "ruby_ui", github: "ruby-ui/ruby_ui", branch: "main"
+gem "ruby_ui", github: "stephannv/ruby_ui", branch: "require_phlex_2"
```
Comment (cirdes) [inline Gemfile:79]: @stephannv , you can now point to ruby_ui again.

**Review by cirdes — COMMENTED** [review]
please fix the ci!

File: Gemfile (hunk 1 above)
Comment (stephannv) [reply Gemfile:79]: I opened this PR https://github.com/ruby-ui/ruby_ui/pull/186 to fix some stimulus problems before pointing to main again.

Comment (stephannv) [reply Gemfile]: I had to adapt some code to work with RubyUI, but I tried to minimize the changes.

**Review by cirdes — APPROVED** [review]

**Pull request merged**